src/SearchParameters/index.js

```diff
@@ -1280,6 +1280,41 @@ SearchParameters.prototype = {
       hierarchicalFacetsRefinements: defaults({}, mod, this.hierarchicalFacetsRefinements)
     });
   },
+
+  /**
+   * Adds a refinement on a hierarchical facet.
+   * @param {string} facet the facet name
+   * @param {string} path the hierarchical facet path
+   * @return {SearchParameter} the new state
+   * @throws Error if the facet is not defined or if the facet is refined
+   */
+  addHierarchicalFacetRefinement: function(facet, path) {
+    if (this.isHierarchicalFacetRefined(facet)) {
+      throw new Error(facet + ' is already refined.');
```

Comment:
> But we may want to add a test to check if it's a hierarchical facet.

It's already done in `isHierarchicalFacetRefined`

> We never threw any errors in similar cases before. Would it make sense to let such double-refinement happen?

True. But it is how we ended up defining those methods in #250